lib/plaid/link_token/user.ex

```diff
@@ -5,13 +5,13 @@ defmodule Plaid.LinkToken.User do
 
   @type t :: %__MODULE__{
           client_user_id: String.t(),
-          legal_name: String.t(),
-          phone_number: String.t(),
-          phone_number_verified_time: String.t(),
-          email_address: String.t(),
-          email_address_verified_time: String.t(),
-          ssn: String.t(),
-          date_of_birth: String.t()
+          legal_name: String.t() | none(),
+          phone_number: String.t() | none(),
+          phone_number_verified_time: String.t() | none(),
+          email_address: String.t() | none(),
+          email_address_verified_time: String.t() | none(),
+          ssn: String.t() | none(),
+          date_of_birth: String.t() | none()
```

Comment:
Since the keys in the map are optional lets use [optional](https://hexdocs.pm/elixir/1.13/typespecs.html#maps) here instead.

```suggestion
          optional(:legal_name) => String.t(),
          optional(:phone_number) => String.t(),
          optional(:phone_number_verified_time) => String.t(),
          optional(:email_address) => String.t(),
          optional(:email_address_verified_time) => String.t(),
          optional(:ssn) => String.t(),
          optional(:date_of_birth) => String.t()
```

Reply:
Going to go ahead and merge this. Thanks for the contribution!

Reply:
Whoops 🤦🏻 `optional` doesn't work in structs. Going to do `String.t() | nil` instead